### PR TITLE
Add Windows support to teleop_twist_keyboard.

### LINKED ISSUE #20: Crashes immediately on Windows
This package relies on the termios and tty python packages, which are not available on Windows.  When attempting to launch the node, I get an immediate error message:

> ```
> C:\Users\dan\Documents\ros2_ws>ros2 run teleop_twist_keyboard teleop_twist_keyboard
> Traceback (most recent call last):
>   File "C:\Users\dan\Documents\ros2_ws\install\teleop_twist_keyboard\lib\teleop_twist_keyboard\teleop_twist_keyboard-script.py", line 11, in <module>
>     load_entry_point('teleop-twist-keyboard', 'console_scripts', 'teleop_twist_keyboard')()
>   File "c:\python37\lib\site-packages\pkg_resources\__init__.py", line 489, in load_entry_point
>     return get_distribution(dist).load_entry_point(group, name)
>   File "c:\python37\lib\site-packages\pkg_resources\__init__.py", line 2843, in load_entry_point
>     return ep.load()
>   File "c:\python37\lib\site-packages\pkg_resources\__init__.py", line 2434, in load
>     return self.resolve()
>   File "c:\python37\lib\site-packages\pkg_resources\__init__.py", line 2440, in resolve
>     module = __import__(self.module_name, fromlist=['__name__'], level=0)
>   File "C:\Users\dan\Documents\ros2_ws\build\teleop_twist_keyboard\teleop_twist_keyboard.py", line 36, in <module>
>     import termios
> ModuleNotFoundError: No module named 'termios'
> ```

### PR DESCRIPTION
**Pull request by clalancette**
Based on the original code from ayushghosh21 in
https://github.com/ros2/teleop_twist_keyboard/pull/23 , but
fixed up to work on both Linux and Windows.

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

This should fix #20

### CODE REVIEW

File: teleop_twist_keyboard.py [hunk 1]
```diff
@@ -99,24 +103,39 @@
 
 
 def getKey(settings):
-    tty.setraw(sys.stdin.fileno())
-    select.select([sys.stdin], [], [], 0)
-    key = sys.stdin.read(1)
-    termios.tcsetattr(sys.stdin, termios.TCSADRAIN, settings)
+    if sys.platform == 'win32':
+        key = chr(ord(msvcrt.getch()))
```
Comment (hidmic) [inline teleop_twist_keyboard.py]: @clalancette this one's curious, why going to an integer and back?

File: teleop_twist_keyboard.py [hunk 2]
```diff
@@ -33,13 +33,17 @@
...
 import sys
-import termios
-import tty
-
-from geometry_msgs.msg import Twist
 
+import geometry_msgs.msg
 import rclpy
 
+if sys.platform == 'win32':
```
Comment (hidmic) [inline teleop_twist_keyboard.py:39]: @clalancette meta: I wonder if it'd make sense to move all console related functions to separate `win32` and non-`win32` modules, and then use this conditional once to `import`

**Review by hidmic — APPROVED** [review]
LGTM

File: teleop_twist_keyboard.py (hunk 1 above)
Comment (clalancette) [reply teleop_twist_keyboard.py]: Hm, good question.  I lifted this code from the original PR, but maybe we can simplify.  I'll give it a shot.

File: teleop_twist_keyboard.py (hunk 2 above)
Comment (clalancette) [reply teleop_twist_keyboard.py:39]: Yeah, I considered that, and also considered having a single module that had all of these `if sys.platform` statements embedded in it, so that `teleop_twist_keyboard.py` wouldn't have to worry about platform-specific differences.

In the end, I decided to do neither because this code is so short.  It's not super hard to follow, so I'd prefer to keep the simplicity of a single file unless this grows significantly more complex.

File: teleop_twist_keyboard.py (hunk 1 above)
Comment (clalancette) [reply teleop_twist_keyboard.py]: I see.  `msvcrt.getch()` returns a byte-string.  We really want `getKey` to return a character.  Thus, we can either do:

* `key = chr(ord(msvcrt.getch()))` (what is already there)
* `key = chr(msvcrt.getch()[0])`

I'm not sure whether one is really preferred over the other.  @hidmic any preferences?

Comment (hidmic) [reply teleop_twist_keyboard.py]: > `msvcrt.getch()` returns a byte-string.

That makes me wonder. Does that mean it is returning a raw code-point in the locale encoding? Shouldn't we treat it as such?

Comment (clalancette) [reply teleop_twist_keyboard.py]: > That makes me wonder. Does that mean it is returning a raw code-point in the locale encoding? Shouldn't we treat it as such?

It's honestly not clear to me.  msvcrt has another method called `getwch`, which returns a Unicode value.  I'll play with it a bit, but the problem is that I don't know how we normalize this with the Linux side of things.  I'm guessing that Linux has a similar issue in that the `read` call is only looking for one byte/character at a time, but I haven't tested non-ASCII there either.

**Review by hidmic — APPROVED** [review]

**Comment by clalancette** [comment]
@hidmic FYI, I ended up changing to use `getwch()` on Windows in 87c6085; that is more consistent with Linux, and more likely to work with non-ASCII keys as well.

One other small change is that I've removed the silly use of `select` in the Linux code; since we were never waiting, and never looking at the ready fds, there was really no reason to do this.

With that said, another review would be appreciated.  Thanks!

**Review by hidmic — APPROVED** [review]
LGTM!
